lib/pure/streams.nim

```diff
@@ -241,19 +244,29 @@ proc readDataStr*(s: Stream, buffer: var string, slice: Slice[int]): int =
     # fallback
     result = s.readData(addr buffer[0], buffer.len)
 
-when not defined(js):
+template jsOrVm(caseJsOrVm, caseElse: untyped): untyped =
+  when nimvm:
+    block:
+      caseJsOrVm
+  else:
```

Comment:
elif defined

Reply:
you cant use expressions or elif branch with `when nimvm`